api/src/main/java/javax/enterprise/event/Event.java

```diff
@@ -100,6 +106,36 @@
 
     /**
      * <p>
+     * Fires an event asynchronously with the specified qualifiers and notifies synchronous and asynchronous observers.
+     * </p>
+     *
+     * @param event the event object
+     * @throws IllegalArgumentException if the runtime type of the event object contains a type variable
+     * @throws FireAsyncException   if a notified observer throws an exception, it will be caught and added
```

Comment:
Yes that's exactly the expected behavior. I'll try to rework that to be more precise on that point.